src/cobra/core/gene.py

```diff
@@ -22,6 +22,12 @@
 from typing import FrozenSet, Iterable, Set, Tuple, Union
 from warnings import warn
 
+# When https://github.com/symengine/symengine.py/issues/334 is resolved, change it to
+# optlang.symbolics.Symbol
+from sympy import Symbol
+from sympy.logic.boolalg import And as sp_And
```

Comment:
Better do somehting like `import sympy.logic.boolalg as spl` and then use `spl.Or` and `spl.And`. Just makes it more explicit where those are coming from.

Reply:
Also check with isort. Looks like that import order would not pass the linter.

Reply:
Ran isort on the modified version